torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -616,14 +616,14 @@ void CudaCodeGen::CompileToNVRTC(
     const std::string& code,
     const std::string& func_name) {
   CUcontext pctx = 0;
-  AT_CUDA_DRIVER_CHECK(nvrtc().cuCtxGetCurrent(&pctx));
   // Note: hacked at::DeviceGuard since at::DeviceGuard was failing to work
   // properly in some scenarios
   const auto prior_device = at::cuda::current_device();
   at::cuda::set_device(this->device().index());
   // cudaSetDevice does not have to really change the underlying device if it
   // doesn't have to, so calling cudaFree to force that change
   CudaSetContext(pctx);
+  AT_CUDA_DRIVER_CHECK(nvrtc().cuCtxGetCurrent(&pctx));
```

Comment:
If you get pctx after CudaSetContext, is CudaSetContext using an uninitialized pctx?

Minor nit: I think given the current structure, it makes sense to inline CudaSetContext to its call site now. Alternatively, merge the "cuCtxGetCurrent" into "CudaSetContext". They could become something like "CudaForceContextInit".

Reply:
Thanks for looking at it. CudaSetContext is called with pctx set to 0 (first line of the function). I'll inline the function for now since it's not used anywhere